sql/core/src/main/scala/org/apache/spark/sql/execution/LocalTableScanExec.scala

```diff
@@ -47,6 +50,22 @@ case class LocalTableScanExec(
 
   private lazy val rdd = sqlContext.sparkContext.parallelize(unsafeRows, numParallelism)
 
+  protected override def doProduce(ctx: CodegenContext): String = {
```

Comment:
Let `LocalTableScanExec` support whole stage codegen.

Because `CollectLimitExec` now supports whole stage codegen, the test in `SQLMetricsSuite`:

    val df2 = spark.createDataset(Seq(1, 2, 3)).limit(2)
    df2.collect()
    val metrics2 = df2.queryExecution.executedPlan.collectLeaves().head.metrics
    assert(metrics2.contains("numOutputRows"))
    assert(metrics2("numOutputRows").value === 2)

will execute the `LocalTableScanExec` node to get its RDD. Then an InputAdapter will connect it to `CollectLimitExec`'s whole stage codegen node. So it will output all 3 rows in the local table.

Adding this whole stage code support seems straightforward. So I adds it here to pass the tests.